server/resources/Fundraiser.py

```diff
@@ -0,0 +1,146 @@
+
+class FundraiserCreate(Resource):
+    """ Handle POST for /fundraiser
+    Create a Fundraiser object
+    User must be logged in to create a Fundraiser
+    Parameters:
+        "title": str,
+        "description": str,
+        "deadline": str (ISO Date format (ex: "2019-12-01T00:00:00")),
+        "amount": int,
+    Autogenerated data:
+        "created_at": str (ISO Date format)
+        "creator_id": int (from User object)
+
+    Returns a Fundraiser object
+    with all the parameter above
+    + an autogenerated id for future reference.
+    + "creator": object of User data of the creator of this Fundraiser
+    + "donations": array of Donation objects related to this Fundraiser
+    """
+
+    @classmethod
+    @jwt_required
+    def post(cls):
+        user_email = get_jwt_identity()
+        user = User.find_by_email(user_email)
+
+        if not user:
+            return {"message": INVALID_CREDENTIALS}, 401
+
+        fundraiser_data = request.get_json()
+
+        deadline_string = fundraiser_data['deadline']
+        deadline_utc = dt.fromisoformat(deadline_string)
+        deadline_utc = deadline_utc.astimezone(tz=tz.utc).isoformat()
+        fundraiser_data['deadline'] = deadline_utc
+
+        # server-side generated data
+        fundraiser_data['created_at'] = dt.now(tz=tz.utc).isoformat()
+        fundraiser_data['last_modified_at'] = dt.now(tz=tz.utc).isoformat()
+        fundraiser_data['creator_id'] = user.id
+
+        try:
+            fundraiser = fundraiser_schema.load(fundraiser_data)
+        except ValidationError as err:
+            return err.messages, 400
+
+        if fundraiser in user.fundraisers:
+            return {"message": FUNDRAISER_EXISTS}, 400
+
+        fundraiser.save_to_db()
+
+        return fundraiser_schema.dump(fundraiser), 200
+
+class FundraiserResource(Resource):
+    """ Handle GET/PUT/DELETE for /fundraiser/<fundraiser_id>
+    PUT/DELETE needs authentication and
+    can only be done by the Fundraiser creator
+    """
+
+    @classmethod
+    def get(cls, fundraiser_id: int):
+        fundraiser = Fundraiser.find_by_id(fundraiser_id)
+
+        if not fundraiser:
+            return {"message": FUNDRAISER_NOT_FOUND}, 404
+
+        return fundraiser_schema.dump(fundraiser), 200
+
+    @classmethod
+    @jwt_required
+    def delete(cls, fundraiser_id:int):
+        user_email = get_jwt_identity()
+        user = User.find_by_email(user_email)
+        fundraiser = Fundraiser.find_by_id(fundraiser_id)
+
+        if not user:
+            return {"message": INVALID_CREDENTIALS}, 401
+
+        if not fundraiser:
+            return {"message": FUNDRAISER_NOT_FOUND}, 404
+
+        if user.id != fundraiser.creator_id:
+            return {"message": INVALID_CREDENTIALS}, 401
+
+        fundraiser.delete_from_db()
+        return {"message": FUNDRAISER_DELETED}, 200
+
+
+    @classmethod
+    @jwt_required
+    def put(cls, fundraiser_id: int):
```

Comment:
@thaumasiotes @aeciorc I'd like your comment on the implementation of the PUT method. It feels repetitive to me. Looking for suggestions from both of you.

Reply:
I see your point, but I'm not sure what to do about it...

Reply:
Seems fine to me, other than combining a couple statements into one-liners, there's not much you can do. On a separate note, are those dict returns converted to json?

Reply:
OK, will leave it as it is. Thanks for the review!

Yep, it's converted to JSON. `flask-restful` is handling the conversion from dict to JSON.